src/snowflake/cli/plugins/nativeapp/manager.py

```diff
@@ -100,6 +104,36 @@ def ensure_correct_owner(row: dict, role: str, obj_name: str) -> None:
         raise UnexpectedOwnerError(obj_name, role, actual_owner)
 
 
+def _get_files_to_sync(paths_to_sync: List[Path], deploy_root: Path) -> List[str]:
+    """Takes a list of paths (files and directories), returning a list of all files recursively, stripping the path to deploy root."""
+    paths = []
+    for path in paths_to_sync:
+        if path.is_dir():
+            for current_dir, _dirs, files in os.walk(path):
+                for file in files:
+                    deploy_path = Path(current_dir, file).relative_to(deploy_root)
+                    paths.append(str(deploy_path))
+        else:
+            paths.append(str(path.relative_to(deploy_root)))
+    return paths
+
+
+def _assert_no_directories(paths_to_sync):
```

Comment:
"assert" is reserved for checking invariants that are never expected to fail, and tends to be disabled in prod (typically for performance). use "verify" here or something similar, as you're checking user input and expect failure as part of normal prod operations (i.e., disabling this check would cause some tests to fail).